setup.py

```diff
@@ -37,7 +37,10 @@ def get_long_description():
     packages=find_packages(exclude=["tests*"]),
     package_data={"starlette": ["py.typed"]},
     include_package_data=True,
-    install_requires=["anyio>=3.0.0,<4"],
+    install_requires=[
+        "anyio>=3.0.0,<4",
+        "typing_extensions; python_version < '3.8'",
```

Comment:
this dep is already in place implicitly via: https://github.com/agronholm/anyio/blob/c90169c0bbf0adc53234bde218e265f1b9157797/setup.cfg#L35-L36